views/topics.js

```diff
@@ -0,0 +1,61 @@
+var View = require('kappa-view-level')
+var EventEmitter = require('events').EventEmitter
+
+module.exports = function (lvl) {
+  var events = new EventEmitter()
+
+  return View(lvl, {
+    map: function (msg) {
+      if (!sanitize(msg)) return []
+      var mappings = []
+      if (msg.value.type === 'chat/topic' && msg.value.content && msg.value.content.channel) {
```

Comment:
OK, so here's the skinny on how views are built:

cabal-core is built on multifeed, which is a set of hypercores. Each hypercore is an append-only log of messages that only one user can write to. multifeed-index iterates over every message in every hypercore, and passes that message to each kappa view, to decide what kind of data structure it would like to build up.

Messages from the p2p database are processed, though not necessarily in any specific order. They *generally* start from sequence number 0 and up to `hypercore.length` for each hypercore, but since messages can arrive out-of-order or even very late, there's no guarantee.

As a result, each view needs to consider both a) how to handle out-of-order messages, and b) how to decide what "latest" means. For example, imagine you set the MOTD for a channel, and then I set it before seeing that you set it. Whose MOTD should be shown? The one with the latest timestamp? What if my clock is wrong and thinks I'm in the year 2025? Then my MOTD is set for the next 7 years! So there's a lot of complexity in deciding "what's the latest X?".

For now, I'm sure this is Good Enough(tm). It just means that some users will see different MOTDs based on what order this view processed its messages.